core/src/main/java/hudson/model/Job.java

```diff
@@ -994,19 +994,8 @@ public RunT getLastCompletedBuild() {
      *   if not enough builds satisfying the threshold have been found. Never null.
      */
     public List<RunT> getLastBuildsOverThreshold(int numberOfBuilds, Result threshold) {
-        
-        List<RunT> result = new ArrayList<>(numberOfBuilds);
-        
         RunT r = getLastBuild();
-        while (r != null && result.size() < numberOfBuilds) {
-            if (!r.isBuilding() && 
-                 (r.getResult() != null && r.getResult().isBetterOrEqualTo(threshold))) {
-                result.add(r);
-            }
-            r = r.getPreviousBuild();
-        }
-        
-        return result;
+        return r.getBuildsOverThreshold(r, numberOfBuilds, threshold);
```

Comment:
IMHO a run object passing itself in into its own member function feels awkward.
Perhaps the function should indicate that it is starting from itself so it wouldn't be so awkward.

Reply:
It can't, in the other case, it doesn't start w/ itself: https://github.com/jenkinsci/jenkins/pull/4259/files#diff-c4f9931d88bca347279b881007d71f0eR957

Reply:
It can call `r.getBuildsOverThreshold` can't it?

Reply:
oh, yeah.

Reply:
ok, so, that work is https://github.com/jsoref/jenkins/pulls/2
And I regret accepting the suggested change, because your suggestion was better (and goes in a different direction).

Reply:
Could you change the method to be `static`? This way we're specifying it's not an object method.

Reply:
@MRamonLeon  https://github.com/jsoref/jenkins/pulls/2 removes the argument. Do you think a static is better?

Reply:
I'm fine with your proposal, it was just another option